src/models/m_rsc.erl

```diff
@@ -362,19 +363,19 @@ merge_delete(WinnerId, LoserId, Context) ->
     m_rsc_update:merge_delete(WinnerId, LoserId, Context).
 
 %% @doc Update a resource
--spec update(resource(), list(), #context{}) -> {ok, resource()} | {error, term()}.
+-spec update(resource(), props(), #context{}) -> {ok, resource()} | {error, term()}.
 update(Id, Props, Context) ->
     m_rsc_update:update(Id, Props, Context).
 
--spec update(resource(), list(), list(), #context{}) -> {ok, resource()} | {error, term()}.
+-spec update(resource(), props(), list(), #context{}) -> {ok, resource()} | {error, term()}.
```

Comment:
The spec in line 370 uses a record, please define a type for the record and use that instead.